src/fe/tdywy.c

```diff
@@ -1072,13 +1072,13 @@ PetscErrorCode TDyComputeErrorNorms_WY(void *context, DM dm, Conditions *conditi
 
 PetscErrorCode TDyUpdateState_WY(void *context, DM dm,
                                  EOS *eos, MaterialProp *matprop,
-                                 CharacteristicCurves *cc, PetscReal *U) {
+                                 CharacteristicCurves *cc,
+                                 PetscInt num_cells, PetscReal *U) {
   PetscErrorCode ierr;
   PetscFunctionBegin;
   TDyWY *wy = context;
 
-  PetscInt cStart, cEnd;
```

Comment:
Are we getting the wrong number of cells here? Why do we have to pass the number of cells to this function?